servicetalk-http-api/src/main/java/io/servicetalk/http/api/HttpLoadBalancerFactory.java

```diff
@@ -29,16 +42,180 @@
      * Converts the passed {@link FilterableStreamingHttpConnection} to a
      * {@link FilterableStreamingHttpLoadBalancedConnection}.
      *
-     * @param connection {@link FilterableStreamingHttpConnection} to convert.
+     * @param connection {@link FilterableStreamingHttpConnection} to convert
+     * @return {@link FilterableStreamingHttpLoadBalancedConnection} for the passed
+     * {@link FilterableStreamingHttpConnection}
+     * @deprecated Implement and use {@link #toLoadBalancedConnection(FilterableStreamingHttpConnection,
+     * ReservableRequestConcurrencyController)}
+     */
+    @Deprecated // FIXME: 0.43 - remove deprecated method
+    default FilterableStreamingHttpLoadBalancedConnection toLoadBalancedConnection(
+            FilterableStreamingHttpConnection connection) {
+        throw new UnsupportedOperationException(
+                "HttpLoadBalancerFactory#toLoadBalancedConnection(FilterableStreamingHttpConnection) " +
+                "is not implemented by " + getClass() + ". This method is deprecated, consider migrating to " +
+                "HttpLoadBalancerFactory#toLoadBalancedConnection(FilterableStreamingHttpConnection, " +
+                "ReservableRequestConcurrencyController) or implement this method if it's required temporarily.");
+    }
+
+    /**
+     * Converts the passed {@link FilterableStreamingHttpConnection} to a
+     * {@link FilterableStreamingHttpLoadBalancedConnection}.
+     *
+     * @param connection {@link FilterableStreamingHttpConnection} to convert
+     * @param concurrencyController {@link ReservableRequestConcurrencyController} to control access to the connection
      * @return {@link FilterableStreamingHttpLoadBalancedConnection} for the passed
-     * {@link FilterableStreamingHttpConnection}.
+     * {@link FilterableStreamingHttpConnection}
      */
-    FilterableStreamingHttpLoadBalancedConnection toLoadBalancedConnection(
-            FilterableStreamingHttpConnection connection);
+    default FilterableStreamingHttpLoadBalancedConnection toLoadBalancedConnection(
+            FilterableStreamingHttpConnection connection,
+            ReservableRequestConcurrencyController concurrencyController) {
```

Comment:
I'm open to discussions here. For now, I tried to keep pre-existing assumption that the `HttpLoadBalancerFactory` knows how to implement `score()` for the passed connection. In practice, the `score()` implementation may not be self-sufficient and can rely on external information from the LB or CF. Since we already pass the `ContextMap` through `request -> selectConnection -> newConnection` chain, we can add it here as a 3rd nullable argument too to preserve that context info. LMK WDYT.

Reply:
Discussed with @tkountis offline and agreed to add the context as the 3rd arg. It may be helpful to propagate information between layers without API changes in the future. Also, consistency with `selectConnection` and `newConnection`.